### PR TITLE
Observability Overview page rework

### PR DESCRIPTION
**Pull request by TackAdam**
### Description

1. Move page options to the header
2. Add UI setting to persist if the cards are disabled
3. Add ability to dismiss the cards to the page header
5. Update the empty state for no dashboard selected
6. Update the fly-out empty state
7. Small UI adjustment to integration table

Disable cards functionality: 

https://github.com/user-attachments/assets/1a283eb6-6c57-4bdf-af68-0c571ad3a42a

| After | Before |
| ------ | ----- |
| Before Image 1 | After Image 1 |
| Before Image 2 | After Image 2 |
| Before Image 3 | After Image 3 |

Update to integrations table
| Before | After |
| ------ | ----- |
| Before Image 1 | After Image 1 |

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TackAdam** [comment]
New cards working with dismiss
Screenshot 2024-10-18 at 9 50 35 AM

File: common/utils/core_services.ts [hunk 1]
```diff
@@ -39,3 +40,4 @@ export const [getOSDSavedObjectsClient, setOSDSavedObjectsClient] = createGetter
   SavedObjectsClientContract
 >('SavedObjectsClient');
 export const [getQueryManager, setQueryManager] = createGetterSetter<QueryManager>('QueryManager');
+export const [getOverviewPage, setOverviewPage] = createGetterSetter<Page>('page');
```
Comment (RyanL1997) [inline common/utils/core_services.ts:43]: nit: "Page"

File: public/components/overview/components/add_dashboard_callout.tsx [hunk 2]
```diff
@@ -3,35 +3,93 @@
...
+            image={
+              <EuiImage
+                src={SelectDashboardSVG}
+                alt="Add a dashboard image"
+                size="m"
+                style={{ objectFit: 'cover', width: '100%', height: '150px' }}
+              />
+            }
+            title="Add a dashboard"
+            description="Customize the overview page by adding a dashboard."
```
Comment (RyanL1997) [inline public/components/overview/components/add_dashboard_callout.tsx:56]: we can add i18n as a follow-up

File: public/components/overview/components/card_configs.tsx [hunk 3]
```diff
@@ -104,4 +104,4 @@ export const cardConfigs = [
   GETTING_STARTED_CONFIG,
   DISCOVER_CONFIG,
   DASHBOARDS_CONFIG,
-];
+];
```
Comment (RyanL1997) [inline public/components/overview/components/card_configs.tsx:107]: lets fix  the lint: 
```
/home/runner/work/dashboards-observability/dashboards-observability/OpenSearch-Dashboards/plugins/dashboards-observability/public/components/overview/components/card_configs.tsx
Error:   107:3  error  Insert `⏎`  prettier/prettier
```

File: public/components/overview/components/select_dashboard_flyout.tsx [hunk 4]
```diff
@@ -86,25 +91,59 @@ export function SelectDashboardFlyout({ closeFlyout, dashboardsSavedObjects, rel
...
-          height="full"
-        >
-          {(list, search) => (
-            <>
-              {search}
-              {list}
-            </>
-          )}
-        </EuiSelectable>
+        {Object.keys(dashboardsSavedObjects).length === 0 ? (
```
Comment (RyanL1997) [inline public/components/overview/components/select_dashboard_flyout.tsx:94]: just for my knowledge, what is this check for? Could you elaborate more of this scenario?

File: public/components/overview/home.tsx [hunk 5]
```diff
@@ -11,60 +19,217 @@ import {
...
+            startDate: '',
+            endDate: '',
+            dashboardTitle: '',
+            dashboardId: '',
+          });
+          ObsDashboardStateManager.isDashboardSelected$.next(false);
         }
+      })
+      .catch((error) => {
+        console.error('Error fetching dashboards:', error);
```
Comment (RyanL1997) [inline public/components/overview/home.tsx:109]: should this be a console error? I know in before we seems to have some console errors on these fetch failures. I think it is okay to leave it here for now, but maybe we can have some better error handling as a follow up.

**Review by RyanL1997 — APPROVED** [review]
Hi @TackAdam , in generally this is LGTM. And I just left some minor comments

File: public/components/overview/components/select_dashboard_flyout.tsx (hunk 4 above)
Comment (TackAdam) [reply public/components/overview/components/select_dashboard_flyout.tsx:94]: If a user does not have a dashboard to set, we want to direct them in a path to create one.
Screenshot 2024-10-18 at 12 25 45 PM
Screenshot 2024-10-18 at 12 26 26 PM

File: common/utils/core_services.ts (hunk 1 above)
Comment (TackAdam) [reply common/utils/core_services.ts]: Done

File: public/components/overview/home.tsx (hunk 5 above)
Comment (TackAdam) [reply public/components/overview/home.tsx:109]: Will look into, thanks for the call out.

**Review by vamsimanohar — APPROVED** [review]

**Pull request merged**